### ISSUE TITLE
[$250] [HOLD for Payment 2024-09-10][Search v2.1] [App] Create In filter

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#46030]
Implement [this section](https://docs.google.com/document/d/1bRV1nPz8SaI_6DHqfehmLnADxv40us_q0JNvmZ1QyaY/edit#bookmark=id.ekk0kmv0drso) of the design doc for the `In` filter only!

cc @adamgrzybowski @Kicu @WojtekBoman @Guccio163

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021833294167898155328</li>
        <li>Upwork Job ID: 1833294167898155328</li>
        <li>Last Price Increase: 2024-09-09</li>
    </ul>
</details>

**Comment by RachCHopkins** [comment]
Already added to project.

**Comment by RachCHopkins** [comment]
@luacmartins do I need to do anything here? Sorry, I'm new to this team.

**Comment by luacmartins** [comment]
Not yet, you'll have to wait until we implement this and the issue is ready for payment.

**Comment by RachCHopkins** [comment]
Changing to monthly in that case!

**Comment by luacmartins** [comment]
The issue is still daily. We're looking for someone from SWM to take this on

**Comment by luacmartins** [comment]
SWM will pick this up once the current wave of filter issues is done.

**Comment by SzymczakJ** [comment]
Sorry for delay, we missed this one. I’d like to work on this issue!

**Comment by SzymczakJ** [comment]
I'll take this after I fix PR comments on https://github.com/Expensify/App/pull/46736. I might also give it to somebody from SWM if I find free person.

**Comment by Guccio163** [comment]
Hi, I'm Wiktor Gut from SWM, I'll handle it so please assign it to me!

**Comment by luacmartins** [comment]
PR is merged, but we still need to address [this issue](https://github.com/Expensify/App/pull/47676#issuecomment-2321534372) in a follow up

**Comment by Guccio163** [comment]
Hi, after research it looks like the filters' wrapping ([issue above](https://github.com/Expensify/App/pull/47676#issuecomment-2321534372)) is a matter of `Text` component, not even `Advanced Filters` one - it is just visible here, because of several `MenuItem`s one under another in a single place. I'll look into it, hopefully it won't turn out to be a big one 🤞
cc: @luacmartins

**Comment by luacmartins** [comment]
@Guccio163 If that's an issue with the text component itself, I'm inclined to create a separate issue for it outside the search project. Would you mind writing a problem statement for it so we can get this issue created while we focus on search functionality?

**Comment by Guccio163** [comment]
@luacmartins Right now I'm consulting with SWM's android developers; I suspect this is a matter of how the shortening is implemented on Android and that their dynamics don't allow developers to enforce ending always exactly on the end of the available space, rather end earlier to not cut words randomly (what IOS supports and in this situation is better for us because it makes Texts aligned better). 

I'll get back to you with possible solutions as soon as I'll be sure of our situation, right now the only possible (sensible) solution is to extend Android Native Text component to provide shortening like we need it, but this seems to be a big piece of work.

**Comment by Guccio163** [comment]
I think I might have found a solution to this problem, I attach iPhone, Android and web comparison below, let me know if a result like this fullfills your expectations:
<img width="1245" alt="Screenshot 2024-09-03 at 11 57 25" src="https://github.com/user-attachments/assets/8a63a3f8-4786-411e-b93f-bb56218e4186">

**Comment by Guccio163** [comment]
**Note:** Our problem solving approach at Expensify is to focus on high value problems and avoid small optimizations with results that are difficult to measure. We also prefer to identify and solve problems at their root. Given that, please ensure all proposed jobs fix a specific problem in a measurable way with evidence so they are easy to evaluate. Here's an example of a good problem/solution:

**Problem:** The `Text` component in `MenuItem` shortens the text too early and inconsistently.

**Solution:** Change `Text` component's `flex` prop to 1, to force using all provided space and thus end in more or less the same place (depending on the content). We can change this behaviour in the whole app or only when using `MenuItem` in `SearchAdvancedFilters`, where the inconsistency is visible the most.

**Comment by luacmartins** [comment]
@Guccio163 nice! Thanks for investigating this. I'm inclined to fix it in the whole App. This might lead to bugs, it's a really simple PR and easy to revert if needed. Let's do it.

**Comment by Guccio163** [comment]
Great, I'm waiting for an issue to link PR with then 😄

**Comment by luacmartins** [comment]
@Guccio163 created [here](https://github.com/Expensify/App/issues/48509)

**Comment by RachCHopkins** [comment]
@luacmartins can I please confirm that this is a standard $250 via Upwork for @rayane-djouah? (@Guccio163 being from SWM)

**Comment by luacmartins** [comment]
Correct

**Comment by RachCHopkins** [comment]
I think I just sent you an offer @rayane-djouah but it did something a little strange - can you please confirm?

**Comment by RachCHopkins** [comment]
Oh, I see - it was because you were super duper fast.

**Comment by RachCHopkins** [comment]
Payment Summary:

Contributor+: @rayane-djouah  paid $250 via Upwork

Upwork job [here](https://www.upwork.com/jobs/~021833294167898155328)

**Comment by RachCHopkins** [comment]
Contributor has been paid, the contract has been completed, and the Upwork post has been closed.
